lib/Sema/CSDiagnostics.cpp

```diff
@@ -683,6 +683,15 @@ bool MissingConformanceFailure::diagnoseAsError() {
     }
   }
 
+  if (isExpr<KeyPathExpr>(anchor)) {
+    auto &ctx = getASTContext();
+    if (protocolType->getAnyNominal() ==
+        ctx.getProtocol(KnownProtocolKind::Copyable)) {
+      emitDiagnostic(diag::expr_keypath_noncopyable_type, nonConformingType);
```

Comment:
What about nonescapable types?

Reply:
We don't have a special diagnostic for those yet, it would be diagnosed as a conformance failure. I can tailed that in a follow-up.

Reply:
I went ahead and added it in this PR since I have to re-run the CI anyway...